### DISCUSSION TITLE
Startup Failure

### DISCUSSION THREAD

**Discussion by bytejunkie** [community/community#26746] [Actions]
<p>I’m getting startup failure on my action, repeatedly.</p>
<p>I’ve used my existing action and I’ve created an action from scratch, no changes from the suggested code. this is how it looks.</p>
<p><img src="https://user-images.githubusercontent.com/4975878/181099708-1261997a-d266-4994-b3f7-121ddbc0a438.png" alt="image" data-base62-sha1="3p5FsvG8FuMQs2PUrjVoC6LZZKS" width="490" height="98"></p>
<p>the most basic i’ve tried is this</p>
<pre><code class="lang-auto">on: workflow_dispatch
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          ref: "main"
</code></pre>
<p>i cant find a single reference to this error message anywhere online. any suggestions would be appreciated!!!</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/bytejunkie">@bytejunkie</a>, welcome to the GitHub Support Community! Is this on a forked repository, or a repository you share with other people? It could be that the GitHub Actions app hasn’t been installed on this repository (something that happens automatically when a workflow is created) and that’s causing this error.</p>
<p>Do you see any other errors listed on the workflow page? And would you be able to post a link to this repository so we can take a closer look?</p>

**Comment by bytejunkie** [comment]
<p>hi <a class="mention" href="/thomasshaped">@thomasshaped</a>,</p>
<p>I dont get any other erros at all.</p>
<p>its my own repo. the action hasnt been run since may and i think the original checkout step has been deprecated since then. it is a bit weird that a brand new workflow wont run when i create it. unless i comment out the checkout action.</p>
<p>i think you’ll see the repo here.</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/bytejunkie/carpics" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars.githubusercontent.com/u/4975878?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/bytejunkie/carpics" target="_blank" rel="noopener nofollow ugc">bytejunkie/carpics</a></h3>


  <p><span class="label1">flask app that deploys to containers. Contribute to bytejunkie/carpics development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<p>
its nothing special, just a container i use to test my docker skills. or not!</p>

**Comment by JessiBrix** [comment]
<p>I am interested in the solution, because I have exactly the same error in my repo. I just startet using github actions, first I tried one workflow on my own. Then I tried the “greetings” example workflow. Both say “Startup failure” without any further information.</p>

**Comment by JessiBrix** [comment]
<p>I fixed this problem by creating a new repo and added only a readme.<br>
Then I used one of the example-actions (in this case it was “Greetings”), copied the master branch from my old repo and edited the action afterwards.<br>
Works fine now.<br>
But I still do not understand why it does not work in my previous repo.</p>

**Comment by bytejunkie** [comment]
<p>did you maybe move your repo from one computer to another? i did (windows to mac) and its about this time that it stopped working.</p>

**Comment by markbirbeck** [comment]
<p>We’ve been having the same problem for about 3 days now. There have not been any changes to the workflows. We can’t get any information out other than the message ‘Startup failed’.</p>
<p>Does anyone know the kind of thing that can cause this? It feels like it could be anything… :frowning:</p>
<p>…are there problems in the workflow files?</p>
<p>…have we hit some kind of quota?</p>
<p>…is there a bug somewhere in the workflow processing?</p>
<p>Any thoughts?</p>
<p>Thanks.</p>

**Comment by billypchan** [comment]
<p>I also encounter this issue when I added the default github action, which only “echo Hello, world!”</p>

**Comment by vaibhavjain-in** [comment]
<p>I am facing the exact same error too. Its been hours I have been trying to see what the issue may be.</p>

**Comment by markbirbeck** [comment]
<p>It looks like it’s related to actions.</p>
<p>Of course that sounds obvious, given that actions are what GHA is all about. :slight_smile:</p>
<p>But my colleague has run a very basic workflow with just an <code>echo</code> and <em>no</em> <code>actions/checkout</code> and that actually works.</p>
<p>And as far as we can tell, it’s the <em>only</em> thing that does.</p>
<p>…so somewhere in the downloading/installing/whatever of those actions, it seems that all is not well… :frowning:</p>

**Comment by markbirbeck** [comment, marked as answer]
<p>Ok…so at least for us, the mystery may be solved. :frowning:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/f/1f48036412e2a5dd37a1f550aa89465c729fce4d.png" data-download-href="/uploads/short-url/4sJ5x56pZt4bKo6aCPGhl1wUWap.png?dl=1" title="Screenshot 2021-01-25 at 19.41.47" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/62572/181099713-d18108ae-2e91-4e62-9306-e58023ad0343.png" alt="Screenshot 2021-01-25 at 19.41.47" data-base62-sha1="4sJ5x56pZt4bKo6aCPGhl1wUWap" width="690" height="418" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/f/1f48036412e2a5dd37a1f550aa89465c729fce4d_2_690x418.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/f/1f48036412e2a5dd37a1f550aa89465c729fce4d_2_1035x627.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/f/1f48036412e2a5dd37a1f550aa89465c729fce4d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/f/1f48036412e2a5dd37a1f550aa89465c729fce4d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-01-25 at 19.41.47</span><span class="informations">1245×756 71.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Reply by gg2** [reply]
I think something gets stuck somehow at a higher level beyond the repository itself -- at the organization level, perhaps. At least in my case.

My action for one particular repo fails with an error "Actions in this workflow must be: within a repository owned by MyOrg."

I just moved a bunch of repositories from Stash to GitHub. The project had its own GitHub organization already, with a repo under it. (It wasn't created new, and I think anything about Organization configuration is irrelevant, but I'm mentioning it just for context.)

The very first new repository I created, I wasn't entirely sure how I wanted to go about transitioning from Stash to GitHub. I started by creating a new repository. Then switched to "Import a repository" using the link given there (no repo created yet). The import from Stash failed, but it auto-created the repository for me, seemingly empty. Adding a new remote for the GitHub repo to my local repo, and doing a "git push --mirror github" worked fine, along with other subsequent revisions to the defined remotes.

I created a set of other new repos in exactly the same way -- except as strictly "new" repos without the "import" step/mistake.

Upon setting up actions in each to mirror to CodeCommit on AWS, the exact same way for each, every other repo except the first one I created worked perfectly. I set the Actions permissions in each to only allow actions from my own Org. But, in doing that, I noticed that the permissions for my 1st repo were the only one set to completely disabled at first. The others started as allowing all.

If I set my 1st repo to Allow all actions from anywhere, the action works.

Since I wanted to avoid one weirdo repo that I'm sure would come back to haunt me by working unpredictably in the future, I deleted the repo; started it new, with the exact same name; redid the whole set up including the action.
And still, the action would not run -- same error -- unless I set actions permissions under settings to allow all, instead of only from my own org.

So, that seems to indicate to me that something got set or stuck outside the repository's own settings -- because of that initial import attempt, I have to guess -- that makes GitHub think the repo doesn't entirely belong to MyOrg. I also tried, on the original rendition of the repo, to transfer ownership. But it didn't list MyOrg, because it seemed to think that the repo did belong to MyOrg.

So, I don't know. But, at least the actions work with permissions set to allow actions from anywhere.

**Comment by otter-computer** [comment]
<p>Sorry for the delay everyone, there was some investigation that took some time to track down what’s going on here. In the mean time it looks like <a class="mention" href="/markbirbeck">@markbirbeck</a> has stumbled onto the actual issue here before I could report back! :laughing:</p>
<p>From the looks of things the pattern here is that you’ve got <code>Allow local actions only</code> set for these repositories. This means that any external actions—even <code>actions/checkout</code>—can’t be included in your workflow.</p>
<p>It looks like a regression has been introduced that has blocked the actual reason being shown in the workflow annotation when a workflow fails startup, instead failing with a vague error. A startup failure could happen for a number of reasons but the most likely one is the Actions permission level being set too low for the steps used in your workflows. I’d recommend double-checking this setting and trying running your workflow again.</p>
<p>I believe the lack of error message should now be fixed though so you should see more details if you’re still having this problem after checking your permissions. :tada:</p>

**Comment by bytejunkie** [comment]
<p>i can confirm that changing the setting highlighted above has worked on my repo too.</p>

**Comment by markbirbeck** [comment]
<p>Thanks <a class="mention" href="/thomasshaped">@thomasshaped</a>.</p>
<p>One other piece of information for you, and anyone else who comes to this thread…</p>
<p>Apparently GH made a change to the default value for the Actions Permissions setting, for enterprise customers. So the setting in our repo changed at the end of last week, but no-one seemed to know about it, and we had no reason to think to check it.</p>
<p>Our Org admins have now changed the setting back for us, so our workflows are running again.</p>
<p>Anyway, if anyone reading this is using GH within an enterprise then it might be worth checking that the Actions permission hasn’t been changed from underneath you.</p>

**Comment by jcs090218** [comment]
<p>I am facing the same issue and tried the option but luck…</p>

**Comment by chalonverse** [comment]
<p>Having the issue for the last hour or so as well. It’s not the option because I switched it to “local actions” and get a different error message. The error I get is just “An unexpected error occurred when executing this workflow.”</p>

**Comment by mayanagarwal** [comment]
<p><a class="mention" href="/jcs090218">@jcs090218</a> <a class="mention" href="/chalonverse">@chalonverse</a> Looks like GitHub Actions is having a problem.<br>
<a href="https://www.githubstatus.com/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://www.githubstatus.com/</a></p>

**Comment by jcs090218** [comment]
<p>I realized that! Thank you!</p>

**Comment by edwinro** [comment]
<p>For those coming to this thread;</p>
<p>I found more information about the startup failure in the workflow file. See Workflows &gt; View workflow file. In my case, it was a syntax error.</p>
<p><img src="https://user-images.githubusercontent.com/16503302/181099728-9cb2fa4b-0004-4d43-a505-d2a29e38ba29.png" alt="image" data-base62-sha1="xw4U922Z4A9yFPvT31RmaHG3Ha6" width="221" height="261"></p>

**Comment by DanielGoldfarb** [comment]
<p>I also encountered a “Startup Failure” <strong>as the result of a syntax error</strong>.  In my case I had spaces in the job name:</p>
<pre><code class="lang-yaml">job name:
  runs-on: ubuntu-latest
  if: ...

  steps:
    - uses: ....
</code></pre>
<p>The above fails with Startup Failure, whereas the below does not fail:</p>
<pre><code class="lang-yaml">job_name:
  runs-on: ubuntu-latest
  if: ...

  steps:
    - uses: ....
</code></pre>
<p><em><strong>It would be nice if github provided more information on a startup failure!</strong></em></p>
